File: iosApp/iosApp/Domain/Models/AnimalSpecies.swift
import Foundation

/// Animal species types supported by the system.
enum AnimalSpecies {
    case dog
    case cat
    case bird
    case rabbit
    case rodent
    case reptile
    case other
    
    /// Failable initializer from string (case-insensitive)
    init?(fromString string: String) {
        let uppercased = string.uppercased()
        switch uppercased {
        case "DOG":
            self = .dog
        case "CAT":
            self = .cat
        case "BIRD":
            self = .bird
        case "RABBIT":
            self = .rabbit
        case "RODENT":
            self = .rodent
        case "REPTILE":
            self = .reptile
        case "OTHER":
            self = .other
        default:
            return nil
        }
    }
}

File: composeApp/src/androidMain/kotlin/com/intive/aifirst/petspot/composeapp/domain/models/AnimalSpecies.kt
package com.intive.aifirst.petspot.composeapp.domain.models

/**
 * Animal species types supported by the system.
 * Determines icon/image display and filtering options.
 */
enum class AnimalSpecies(val displayName: String) {
    DOG("Dog"),
    CAT("Cat"),
    BIRD("Bird"),
    RABBIT("Rabbit"),
    OTHER("Other"),
}
